File: app/src/main/java/io/blackbox_vision/helpers/ui/fragment/DetailsFragment.java
package io.blackbox_vision.helpers.ui.fragment;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import butterknife.BindView;

import io.blackbox_vision.helpers.R;
import io.blackbox_vision.helpers.logic.factory.DetailsPresenterFactory;
import io.blackbox_vision.helpers.util.MockUtils;
import io.blackbox_vision.helpers.logic.presenter.DetailsPresenter;
import io.blackbox_vision.helpers.logic.view.DetailsView;
import io.blackbox_vision.mvphelpers.logic.factory.PresenterFactory;
import io.blackbox_vision.mvphelpers.ui.fragment.BaseFragment;


public final class DetailsFragment extends BaseFragment<DetailsPresenter> implements DetailsView {

    @BindView(R.id.sample_button)
    Button button;

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        button.setOnClickListener(v -> {
            if (null != getPresenter()) {
                getPresenter().findRequiredInformation(MockUtils.SAMPLE);
            }
        });
    }

    @NonNull
    @Override
    public PresenterFactory<DetailsPresenter> createPresenterFactory() {
        return DetailsPresenterFactory.newInstance();
    }

    @LayoutRes
    @Override
    public int getLayout() {
        return R.layout.fragment_details;
    }

    @Override
    public void onPresenterCreated(@NonNull DetailsPresenter presenter) {
        presenter.attachView(this);
    }

    @Override
    public void onInfoReceived(@NonNull Bundle information) {
        Toast.makeText(getContext(), "This is the information ---> " + information.get(MockUtils.SAMPLE), Toast.LENGTH_LONG).show();
    }

    @Override
    public void onInfoError(@NonNull String errorMessage) {
        Toast.makeText(getContext(), "This is the error ---> " + errorMessage, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/io/blackbox_vision/helpers/ui/fragment/DetailsFragment.kt
package io.blackbox_vision.helpers.ui.fragment

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.view.View
import android.widget.Button
import android.widget.Toast

import butterknife.BindView

import io.blackbox_vision.helpers.R
import io.blackbox_vision.helpers.util.MockUtils
import io.blackbox_vision.helpers.logic.presenter.DetailsPresenter
import io.blackbox_vision.helpers.logic.view.DetailsView
import io.blackbox_vision.mvphelpers.ui.fragment.BaseFragment


class DetailsFragment : BaseFragment<DetailsPresenter>(), DetailsView {

    @BindView(R.id.sample_button)
    internal lateinit var button: Button

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        button.setOnClickListener { v -> presenter!!.findRequiredInformation(MockUtils.SAMPLE) }
    }

    override fun addPresenter(): DetailsPresenter {
        return DetailsPresenter()
    }

    override val layout: Int
        @LayoutRes
        get() = R.layout.fragment_details

    override fun onPresenterCreated(presenter: DetailsPresenter) {
        presenter.attachView(this)
    }

    override fun onInfoReceived(information: Bundle) {
        Toast.makeText(context, "This is the information ---> ${information.get(MockUtils.SAMPLE)!!}", Toast.LENGTH_LONG).show()
    }

    override fun onInfoError(errorMessage: String) {
        Toast.makeText(context, "This is the error ---> $errorMessage", Toast.LENGTH_SHORT).show()
    }
}
